src/device-state/current-state.ts

```diff
@@ -104,170 +109,149 @@ const report = async ({ stateDiff, conf }: StateReport) => {
+				reportHandler,
+				// Do not do exponential backoff if the API reported a retryAfter
+				e.retryAfter ? 0 : attempts,
+				maxDelay,
+				// Start the polling at the value given by the API if any
+				e.retryAfter ?? patchInterval,
+			);
+		} else {
+			// Report did not get a HTTP response
+			eventTracker.track('Device state report failure', {
+				error: e.message,
+			});
+			await backoffPromise(
+				reportHandler,
+				stateReportErrors++,
+				maxDelay,
+				patchInterval,
+			);
+		}
+	}
 
-		// report state diff
-		throttledReport(
-			{ stateDiff, conf },
-			() => {
-				// If the patch succeeds update lastReport and reset the error counter
-				_.assign(lastReportedState.local, stateDiff.local);
-				_.assign(lastReportedState.dependent, stateDiff.dependent);
-				stateReportErrors = 0;
+	// Report was successful so update lastReportedState
+	_.assign(lastReportedState.local, stateDiff.local);
+	_.assign(lastReportedState.dependent, stateDiff.dependent);
+	stateReportErrors = 0;
 
-				reportPending = false;
-			},
-			(e) => {
-				if (e instanceof StatusError) {
-					// We don't want these errors to be classed as a report error, as this will cause
-					// the watchdog to kill the supervisor - and killing the supervisor will
-					// not help in this situation
-					log.error(
-						`Device state report failure! Status code: ${e.statusCode} - message:`,
-						e.message,
-					);
+	// Wait for interval before trying to report again
+	await sleep(patchInterval);
 
-					// We want to backoff on all errors, but without having the healthchecks
-					// get triggered.
-					// This will use retryAfter as maxDelay if the header is present in the
-					// response by the API
-					backoff(
-						reportCurrentState,
-						// Do not do exponential backoff if the API reported a retryAfter
-						e.retryAfter ? 0 : attempts,
-						maxDelay,
-						// Start the polling at the value given by the API if any
-						e.retryAfter ?? MINIMUM_BACKOFF_DELAY,
-					);
-				} else {
-					eventTracker.track('Device state report failure', {
-						error: e.message,
-					});
-					backoff(reportCurrentState, stateReportErrors++, maxDelay);
-				}
-			},
-		);
-	})();
+	// Begin report loop again with 0 attempts
+	return reportHandler(0);
 }
 
-export const startReporting = () => {
-	const doReport = () => {
-		if (!reportPending) {
-			reportCurrentState();
-		}
-	};
-
-	// If the state changes, report it
-	deviceState.on('change', doReport);
-	// But check once every max report frequency to ensure that changes in system
-	// info are picked up (CPU temp etc)
-	setInterval(doReport, constants.maxReportFrequency);
-	// Try to perform a report right away
-	return doReport();
+export const startReporting = async (): Promise<void> => {
+	// Set config values we need to avoid multiple db hits
+	patchInterval = await config.get('statePatchInterval');
+	// All config values fetch, start reporting!
+	return reportHandler();
```

Comment:
Might be worth discussing the consequences of removing the device state change listener which provides more immediate updates to the dashboard.

Reply:
I'd definitely lean towards keeping the immediate report check, if the device hasn't reported anything recently then showing it's started updating asap makes sense imo. In general the behavior should be one of reporting at most once every X seconds rather than only reporting when hitting the X seconds interval. It's a somewhat minor difference but keeping that first update happening immediately if it's otherwise quiet can help the UX a lot imo without adding additional load